pkg/promtail/api/entry_parser.go

```diff
@@ -56,6 +67,20 @@ func (e *EntryParser) UnmarshalYAML(unmarshal func(interface{}) error) error {
 // Wrap implements EntryMiddleware.
 func (e EntryParser) Wrap(next EntryHandler) EntryHandler {
 	switch e {
+	case CRI:
+		return EntryHandlerFunc(func(labels model.LabelSet, _ time.Time, line string) error {
+			parts := criPattern.FindStringSubmatch(line)
+			if parts == nil || len(parts) < 5 {
+				return fmt.Errorf("Line did not match the CRI log format")
+			}
+
+			timestamp, err := time.Parse(time.RFC3339Nano, parts[1])
+			if err != nil {
+				return fmt.Errorf("CRI timestamp '%s' does not match RFC3339Nano", parts[1])
+			}
+
+			return next.Handle(labels, timestamp, parts[4])
```

Comment:
It seems like stderr/stdout should be added to `labels` so that we can distinguish between the 2 streams.

Reply:
I did that originally as the docker format did, but it added an extra stream:stdout label visible in grafana that I thought cluttered it and wasn't there in raw. But sure I can add it back.

Reply:
@DanCech Done